<?xml version="1.0"?>
<launch>

    <!-- these are the arguments you can pass this launch file -->

    <arg name="x" default="0"/>
    <arg name="y" default="0"/>
    <arg name="Y" default="0"/>

    <!-- Parameters for the local planner-->
    <arg name="random_spawn_dyn_obj" default="false"/>
    <arg name="random_waypoint_generation" default="false"/>

    <!-- robot -->
    <arg name="robot_x" default="277116.461699"/>
    <arg name="robot_y" default="4683499.62845"/>
    <arg name="robot_Y" default="0"/>
    <arg name="robot_x_end" default="277116.46169"/>
    <arg name="robot_y_end" default="4683499.62845"/>
    <arg name="robot_Y_end" default="0"/>




    <!-- Combine joint values-->
<!--    <group ns="sherpa"> -->

        <!-- Gui Parameter -->
        <param name="use_gui" value="false"/>

        <arg name="tf_prefix" value="rbsherpa_hl_"/>
        <arg name="robot_name" default="rbsherpa_hl_"/>


    <!-- <node name="mid_cont" pkg="sherpa_ros" type="sherpa_mid_cont_node"/> -->
    <node name="map_manager" pkg="sherpa_ros" type="map_manager_node"/>
  
   <include file="$(find rm3_ackermann_controller)/launch/ackermann_controller.launch">
    <arg name="lyapunov_enable" value="true"/>
    <arg name="mode" value="5"/>
    <arg name="error" value="0"/>
   </include>

    <node name="ibvs_controller_node" pkg="rvb_mpc" type="IBVS_random_controller_node" clear_params="true" output="screen" 
          args="$(find rvb_mpc)/parameters/params_short_term.yaml">
    </node>


    <node name="global_planner" pkg="sherpa_ros" type="global_planner.py" output="screen">
            <param name="tour" value="tour1" />
            <param name="origin_x" value="$(arg robot_x_end)" />
            <param name="origin_y" value="$(arg robot_y_end)" />
            <param name="origin_Y" value="$(arg robot_Y_end)" />
<!--            <remap from="/waypoint" to="/$(arg robot)/waypoint" />-->
<!--            <remap from="/odom_gps" to="/$(arg robot)/odom_gps" />-->
        </node>



</launch>
